### PR TITLE
RGW NFS: add nfs.rst to doc/radosgw

### PR DESCRIPTION
**Pull request by mattbenjamin**
This document contains sections by John Wilkins <jowilkin@redhat.com>.

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>

### CODE REVIEW

File: doc/radosgw/nfs.rst [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+<ganesha-host-name>:/ <mount-point> nfs noauto,soft,nfsvers=4.1,sync,proto=tcp 0 0
+
+Specify the NFS-Ganesha host name and the path to the mount point on
+the client. 
+
+References
+==========
+
+[1] http://docs.aws.amazon.com/AmazonS3/latest/dev/ListingKeysHierarchy.html
```
Comment (cbodley) [inline doc/radosgw/nfs.rst]: there's a special format you can use for links like this. an example from https://raw.githubusercontent.com/ceph/ceph/master/doc/radosgw/encryption.rst:
```
This is implemented in S3 according to the `Amazon SSE-KMS`_ specification.
```
then at the bottom:
```
.. _Amazon SSE-KMS: http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html
```

Comment (cbodley) [inline doc/radosgw/nfs.rst]: alternatively, there's a [footnote syntax](http://docutils.sourceforge.net/docs/user/rst/quickref.html#footnotes)

File: doc/radosgw/nfs.rst [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+* Links, including symlinks, are not supported
+* Directories may not be moved/renamed
+  * files may be moved between directories
+* Only full, sequential *write* i/o is supported
+
+ - i.e., write operations are constrained to be **uploads**
+ - many typical i/o operations such as editing files in place will necessarily fail as they perform non-sequential stores
+ - some file utilities *apparently* writing sequentially (e.g., some versions of GNU tar) may fail due to infrequent non-sequential stores
+ - When mounting via NFS, sequential application i/o can generally be constrained to be written sequentially to the NFS server via a synchronous mount option (e.g. -osync in Linux)
+ - NFS clients which cannot mount synchronously (e.g., MS Windows) will not be able to upload files
```
Comment (cbodley) [inline doc/radosgw/nfs.rst]: it doesn't appear to like the list formatting. github will show you a preview: https://github.com/linuxbox2/ceph/blob/10ea9a515784ce1a4bae55ef4fd257d2215125f9/doc/radosgw/nfs.rst

File: doc/radosgw/nfs.rst [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+Required ceph.conf configuration for RGW NFS includes:
+
+* valid [client.radosgw.{instance-name}] section
+* valid values for minimal instance configation, in particular, an installed and correct *keyring*
+
+Other config variables are optional, front-end-specific and front-end
+selection variables (e.g., *rgw data* and *rgw frontends*) are
+optional and in some cases ignored.
+
+A small number of config variables (e.g., *rgw_namespace_expire_secs*)
```
Comment (cbodley) [inline doc/radosgw/nfs.rst]: other radosgw docs use the ` ``double-backtick`` ` syntax to highlight config variables, config sections, and other terms. see https://raw.githubusercontent.com/ceph/ceph/master/doc/radosgw/config-ref.rst for examples

Comment (mattbenjamin) [inline doc/radosgw/nfs.rst]: ok

**Comment by mattbenjamin** [comment]
@cbodley updated

File: doc/radosgw/nfs.rst [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+NFS-Ganesha-specific configuration in the NFS-Ganesha config file,
+ganesha.conf.
+
+ceph.conf
+---------
+
+Required ceph.conf configuration for RGW NFS includes:
+
+* valid [client.radosgw.{instance-name}] section
+* valid values for minimal instance configation, in particular, an installed and correct ``keyring``
```
Comment (ricardoasmarques) [inline doc/radosgw/nfs.rst]: **typo:** s/configation/configuration/

File: doc/radosgw/nfs.rst [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+   {
+        Export_ID={numeric-id};
+        Path = "/";
+        Pseudo = "/";
+        Access_Type = RW;
+        SecType = "sys";
+        NFS_Protocols = 4;
+        Transport_Protocols = TCP;
+
+	# optional, permit unsquashed access by client "root" user
```
Comment (ricardoasmarques) [inline doc/radosgw/nfs.rst]: This line should use space indentation instead of tab indentation

**Comment by mattbenjamin** [comment]
@ricardoasmarques thank you!

**Pull request merged**